active/NumberBox/NumberBox.md

```diff
@@ -0,0 +1,320 @@
+# Number box
+
+## Background
+Xaml has a TextBox control for text input, which can be used for numeric input, but numeric input scenarios benefit from custom features, such as buttons to increment/decrement. The NumberBox control in this spec provides these features. This will ship as part of the [WinUI package](https://www.nuget.org/packages/Microsoft.UI.Xaml), not as part of the Windows OS.
+
+## Description
+Represents a control that can be used to display and edit numbers. This supports validation, increment stepping, and computing inline calculations of basic equations, such as multiplication, division, addition, and subtraction.
+
+## Is this the right control? 
+
+You can use a NumberBox control to capture and display mathematic input. If you need an editable text box that accepts more than numbers, use the [TextBox](https://docs.microsoft.com/uwp/api/Windows.UI.Xaml.Controls.TextBox) control. If you need an editable text box that accepts passwords or other sensitive input, see [PasswordBox](https://docs.microsoft.com/uwp/api/windows.ui.xaml.controls.passwordbox). If you need a text box to enter search terms, see [AutoSuggestBox](https://docs.microsoft.com/uwp/api/windows.ui.xaml.controls.autosuggestbox). If you need to enter or edit formatted text, see [RichEditBox](https://docs.microsoft.com/uwp/api/windows.ui.xaml.controls.richeditbox).
+
+## Examples
+
+### Create a simple NumberBox
+
+Here's the XAML for a basic NumberBox that demonstrates the default look. Use [x:Bind](https://docs.microsoft.com/en-us/windows/uwp/xaml-platform/x-bind-markup-extension#property-path) to ensure the data displayed to the user remains in sync with the data stored in your app. 
+
+
+```XAML
+<NumberBox Value="{x:Bind Path=ViewModel.NumberBoxValue, Mode=TwoWay}" />
+```
+![An in-focus input field showing 0.](images/numberbox-basic.PNG)
+
+### Labeling NumberBox
+
+Use `Header` or `PlaceholderText` if the purpose of the NumberBox isn't clear. `Header` is visible whether or not the NumberBox has a value. 
+
+```XAML
+<NumberBox Header="Enter expression:"
+    Value="{x:Bind Path=ViewModel.NumberBoxValue, Mode=TwoWay}" />
+```
+
+![A header reading "Enter expression:" above a NumberBox.](images/numberbox-header.PNG)
+
+`PlaceholderText` is displayed inside the NumberBox and disappears once a value has been entered.
+
+```XAML
+<NumberBox PlaceholderText="A + B"
+    Value="{x:Bind Path=ViewModel.NumberBoxValue, Mode=TwoWay}" />
+```
+
+![A NumberBox containing placeholder text reading "A + B".](images/numberbox-placeholder-text.PNG)
+
+### Enable calculation support
+
+Setting the `AcceptsCalculation` property to true enables NumberBox to evaluate basic inline expressions such as multiplication, division, addition, and subtraction using standard order of operations. Evaluation is triggered on loss of focus or when the user presses the "Enter" key. Once an expression is evaluated, the original form of the expression is not preserved.
```

Comment:
`Accepts-` is a rare prefix in this namespace.  Both RichEdit and TextBox have `AcceptsReturn` which I understand means "the box supports multiline input."  There's no "Calculation" key or action.  Should this instead be a property indicating when calculation is performed? Some choices include "Never" or "WhenFocusLost" or "WhenReturnPressed"?

Reply:
Discussion: Does this relate to validation? About validation ... Don't want to block non-numeric when AcceptsCalculation is false. Right now if you type something invalid, it will revert back to the value prior to the invalid calculation. There are plans for additional available behaviors in WinUI 3.0.

AcceptsCalculation
CalculatesOnCommit
CalculationAllowed
AllowCalculation
AllowsInlineCalculation
IsExpressionInputEnabled
CanEnterExpression
AcceptsExpression

Recommend: AcceptsExpression

Reply:
Resolved.

Reply:
Resolved with [this ](https://github.com/microsoft/microsoft-ui-xaml-specs/pull/52/commits/98f62ea06212dd8a2224d4635fb664bec76dc7d4)commit.